tests/test_classification.py

```diff
@@ -779,3 +857,620 @@ def test_sampling_for_multinomialNB():
+                           'log': output_dir,
+                           'models': output_dir,
+                           'results': output_dir,
+                           'predictions': output_dir,
+                           'probability': 'true' if use_probabilities else 'false',
+                           'metrics': "['{}']".format(metric_name)}
+
+    config_path = fill_in_config_options(config_template_path,
+                                         values_to_fill_dict,
+                                         '{}_{}'.format(metric_name, use_probabilities),
+                                         good_probability_option=True)
+
+    # run this experiment and load the results_json and the SKLL model
+    results_json_path = run_configuration(config_path, local=True, quiet=True)[0]
+    results_obj = json.load(open(results_json_path, 'r'))[0]
+    model_file_path = join(output_dir,
+                           '{}_metric_{}.model'.format(experiment_name,
+                                                       results_obj['learner_name']))
+    clf = Learner.from_file(model_file_path)
+
+    # get the value of the metric from SKLL
+    skll_metric_value = results_obj['additional_scores'][metric_name]
+
+    # get the feature matrix (X) and the labels array (y) for both
+    # the training and test set to pass to scikit-learn; note that
+    # we want y to be class indices since those are what we need
+    # when computing the metrics to match what SKLL does
+    X_train = train_fs.features
+    y_train = [clf.label_dict[label] for label in train_fs.labels]
+    X_test = test_fs.features
+    y_test = [clf.label_dict[label] for label in test_fs.labels]
+
+    # instantiate a LogisticRegression models with the default
+    # parameters that are used in SKLL
+    model_kwargs = {'max_iter': 1000,
+                    'solver': 'liblinear',
+                    'multi_class': 'auto',
+                    'random_state': 123456789}
+
+    sklearn_learner = LogisticRegression(**model_kwargs)
+
+    # now generate predictions on the test set from sklearn and
+    # then compute the metrics appropriately (using either the
+    # labels or the probabilities) and store them for comparison
+    # for training, we use the class indices rather than the class
+    # labels like SKLL does so that our predictions are in the index
+    # space rather than label space
+    sklearn_learner.fit(X_train, y_train)
+
+    # compute both labels and probabilities via sklearn
+    sklearn_test_labels = sklearn_learner.predict(X_test)
+    sklearn_test_probs = sklearn_learner.predict_proba(X_test)
+
+    # now let's proceed on a metric by metric basis
+    # and note that we are again using the class
+    # indices rather than the class labels themselves
+
+    # 1. Correlation metrics are only computed for integer
+    #    or float labels; they use probability values if
+    #    available only in the binary case.
+    if metric_name == 'pearson':
```

Comment:
Same comment as above.

Reply:
I don't want to do the refactoring here because I don't want to run as little custom SKLL code as possible in a test.

Reply:
I don't want to do any refactoring here as I want to run as little SKLL code during tests as possible.

Reply:
I'm not sure I get what you mean. I just meant that you could have something like the following:
```python
if metric_name in ['pearson', 'spearman', 'kendaulltau']:
    if metric_name == 'pearson':
        metric_func = pearsonr
    elif metric_name == 'spearman':
        metric_func = spearmanr
    else:
        metric_func = kendalltau
    if issubclass(label_type, (np.int64, np.float64)):
        if len(label_array) == 2 and use_probabilities:
            sklearn_metric_value = metric_func(y_test,
                                            sklearn_test_probs[:, 1])[0]
        else:
            sklearn_metric_value = metric_func(y_test,
                                            sklearn_test_labels)[0]
```

It's a little clunky now that I see it, but I think it does reduce duplication a bit.

Reply:
Oh I see what you mean! That's a good idea 👍.